src/Microsoft.SqlTools.ServiceLayer/ObjectManagement/ObjectTypes/Server/ServerData.cs

```diff
@@ -0,0 +1,805 @@
+﻿//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+#nullable disable
+
+using System;
+using Microsoft.SqlServer.Management.Smo;
+using Microsoft.SqlTools.ServiceLayer.Management;
+using Microsoft.SqlServer.Management.Common;
+using Microsoft.SqlTools.ServiceLayer.ServerConfigurations;
+
+namespace Microsoft.SqlTools.ServiceLayer.ObjectManagement
+{
+    /// <summary>
+    /// ServerGeneral - main app server page
+    /// </summary>
+    internal class ServerPrototype
+    {
+        #region Members
+
+        /// <summary>
+        /// data container member that contains data specific information like
+        /// connection infor, SMO server object or an AMO server object as well
+        /// as a hash table where one can manipulate custom data
+        /// </summary>
+        private CDataContainer dataContainer = null;
+        private ServerConnection sqlConnection = null;
+        private ServerConfigService configService = null;
+
+
+        private ServerPrototypeData currentState;
+        private ServerPrototypeData originalState;
+        #endregion
+
+        #region Trace support
+        private const string componentName = "ServerGeneral";
```

Comment:
Why ServerGeneral?

Reply:
It was for general properties but I'm including memory settings as well. So, I removed it.